src/test/ui/consts/miri_unleashed/const_refers_to_static_cross_crate.stderr

```diff
@@ -4,38 +4,46 @@ error[E0080]: it is undefined behavior to use this value
 LL | / const SLICE_MUT: &[u8; 1] = {
 LL | |
 LL | |
+LL | |
 LL | |     unsafe { &static_cross_crate::ZERO }
 LL | | };
    | |__^ type validation failed: encountered a reference pointing to a static variable
    |
    = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rustc repository if you believe it should not be considered undefined behavior.
+   = note: the raw bytes of the constant (size: 8, align: 8) {
+               ╾───────alloc0────────╼                         │ ╾──────╼
+           }
 
 error: could not evaluate constant pattern
-  --> $DIR/const_refers_to_static_cross_crate.rs:39:9
+  --> $DIR/const_refers_to_static_cross_crate.rs:41:9
    |
 LL |         SLICE_MUT => true,
    |         ^^^^^^^^^
 
 error[E0080]: it is undefined behavior to use this value
-  --> $DIR/const_refers_to_static_cross_crate.rs:17:1
+  --> $DIR/const_refers_to_static_cross_crate.rs:18:1
    |
 LL | / const U8_MUT: &u8 = {
 LL | |
 LL | |
+LL | |
 LL | |     unsafe { &static_cross_crate::ZERO[0] }
 LL | | };
    | |__^ type validation failed: encountered a reference pointing to a static variable
    |
    = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rustc repository if you believe it should not be considered undefined behavior.
+   = note: the raw bytes of the constant (size: 8, align: 8) {
+               ╾───────alloc0────────╼                         │ ╾──────╼
```

Comment:
It would be nice to show the static name instead of alloc id.

Reply:
that can often be problematic since the path of a static can be very long. But it would be nice to improve something in this situation, indeed.

Reply:
With a failed length check, a "glossary" can be added as a separate note (`note: alloc0 is `Very::Long::Type::Name`).

Reply:
We do have a scheme for recursively printing allocaitons, which will then also tell us what static item is being pointed at. This is where that happens during MIR printing: https://github.com/rust-lang/rust/blob/704e47f78b4c8801a3c76f235a5a152e1b60b300/compiler/rustc_mir/src/util/pretty.rs#L641

in the spirit of code deduplication we should split that function into the current one taking a `&Body` and another one that takes a `Vec<AllocId>` or sth that gets called by the first one. Then the diagnostic code from this PR can use the new function and pass a `vec![alloc_id]`.

That said... I think we should first merge this PR and then keep the issue open to improve the output further. This is already a huge improvement